### PR TITLE
Breadcrumbs

### PR DESCRIPTION
**Pull request by crod951**
Adding breadcrumbs route, helper and tests

Review and Merge after #48 is merged into main

### CODE REVIEW

**Review by akinsey — APPROVED** [review]
Looks good

File: test/epochtalk_server_web/controllers/breadcrumb_test.exs [hunk 1]
```diff
@@ -0,0 +1,272 @@
+defmodule Test.EpochtalkServerWeb.Controllers.Breadcrumb do
+  use Test.Support.ConnCase, async: true
+  import Test.Support.Factory
+  alias EpochtalkServer.Models.User
+  @test_username "user"
+
+  setup %{conn: conn} do
```
Comment (unenglishable) [inline test/epochtalk_server_web/controllers/breadcrumb_test.exs:7]: %{conn: conn, users: %{user: user}}

File: test/epochtalk_server_web/controllers/breadcrumb_test.exs [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    build(:board_mapping,
+      attributes: [
+        [category: category, view_order: 0],
+        [board: parent_board, category: category, view_order: 1],
+        [board: child_board, parent: parent_board, view_order: 2],
+        [board: board_no_parent, category: category, view_order: 3]
+      ]
+    )
+
+    {:ok, user} = User.by_username(@test_username)
```
Comment (unenglishable) [inline test/epochtalk_server_web/controllers/breadcrumb_test.exs:22]: see above

File: test/epochtalk_server_web/controllers/breadcrumb_test.exs [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    threads = build_list(1, :thread, board: child_board, user: user)
```
Comment (unenglishable) [inline test/epochtalk_server_web/controllers/breadcrumb_test.exs:23]: build(:thread, board: child_board, user: user)

File: test/epochtalk_server_web/controllers/breadcrumb_test.exs [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+    threads_no_parent = build_list(1, :thread, board: board_no_parent, user: user)
```
Comment (unenglishable) [inline test/epochtalk_server_web/controllers/breadcrumb_test.exs:24]: same as above, don't need to use build_list

File: test/epochtalk_server_web/controllers/breadcrumb_test.exs [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+      response =
+        conn
+        |> get(
+          Routes.breadcrumb_path(conn, :breadcrumbs, %{"id" => category.id, "type" => "category"})
+        )
+        |> json_response(200)
+
+      assert Enum.count(response["breadcrumbs"]) == 1
+
+      response_category_crumb = response["breadcrumbs"] |> Enum.at(0)
```
Comment (unenglishable) [inline test/epochtalk_server_web/controllers/breadcrumb_test.exs:52]: Enum.first

**Review by unenglishable — COMMENTED** [review]
looks good, functionality is there, could use a little refactoring

**Pull request merged**